schema-engine/connectors/sql-schema-connector/src/introspection_pair/model.rs

```diff
@@ -47,6 +47,16 @@ impl<'a> ModelPair<'a> {
         self.next.has_row_level_security()
     }
 
+    /// Whether the model has check constraints.
+    pub(crate) fn has_check_constraints(self) -> bool {
+        self.next.has_check_constraints()
```

Comment:
This will repeat the comment in re-introspection. You have to check that `self.previous.is_none()` to prevent that.